### ISSUE TITLE
Debugger loading incorrect source file or throwing "No source file" error

### ISSUE THREAD

**Issue by kevinbluer** [trufflesuite/vscode-ext#54]
Repo steps as follows:

- `truffle unbox amm` (or equivalent via the extension itself)
- `truffle migrate`
- CMD+Shift+P | Debug Transaction

![Screen Shot 2022-04-06 at 2 56 03 PM](https://user-images.githubusercontent.com/210755/162060139-33c74181-4eff-4fbc-a665-fc755c85f484.png)

Selecting the first txn (`CSAMM.constructor()`) resulting in the debugger incorrectly jumping to `Token1.sol` and the second results in `Error occured in debug mode: No source file`.

![Screen Shot 2022-04-06 at 2 57 28 PM](https://user-images.githubusercontent.com/210755/162060630-a2777e0d-2eaa-4966-9e7b-a7197be2a844.png)

Note that debugging either of these via the CLI is fine.

**Comment by michaeljohnbennett** [comment]
I have been testing this and it seems something inside the debugger with these specific files. 

I can't figure out why the debugger doesn't like the source or finds the source correctly for this file.

**Comment by xhulz** [comment]
I've reproduced the steps above and when I was debugging the Token1.sol, I got this error stack. I realized something strange, it's trying to find a file located: **Users/tyler/projects/truffle/packages/debugger/lib/data/selectors/index.js**

`TypeError: Cannot read properties of undefined (reading 'variables')

rejected promise not handled within 1 second: TypeError: Cannot read properties of undefined (reading 'variables')

stack trace: TypeError: Cannot read properties of undefined (reading 'variables')
    at map (/Users/tyler/projects/truffle/packages/debugger/lib/data/selectors/index.js:676:54)
    at Array.map (<anonymous>)
    at map (/Users/tyler/projects/truffle/packages/debugger/lib/data/selectors/index.js:675:54)
    at Array.map (<anonymous>)
    at /Users/tyler/projects/truffle/packages/debugger/lib/data/selectors/index.js:658:37
    at /Users/marcosschulz/Documents/Code/Consensys/vscode-ext/node_modules/reselect/lib/index.js:87:25
    at memoized (/Users/marcosschulz/Documents/Code/Consensys/vscode-ext/node_modules/reselect/lib/defaultMemoize.js:135:20)`

**Comment by xhulz** [comment]
I'm talking to Tyler to understand why it’s happening

**Comment by michaeljohnbennett** [comment]
its hard to debug the compressed sourcemap dependencies of debugger. Locally building a development/sourcemap version of debugger and using `yarn link ...` will get you further to seeing the issue. The tyler thing is a red herring, its the way webpack builds the file names. I fixed it somewhat a few months ago but these paths are not the real paths the files are looking for but actually the referenced files inside the sourcemaps generated inside the debugger package as part of the release process.

TLDR; You can't get meaningful debug information from the debugger, the sagas/redux and `*yield` calls with the webpacked prod code. You need to make a `development` build and link it in and run locally.

**Comment by xhulz** [comment]
cc @acuarica for visibility

**Comment by pupudaye** [comment]
=============================================================
The truffle development case "metacoin" can be debugged normally, but it can't be debugged properly when switching to complex projects that rely on contracts.

The steps are not easy to express clearly, or can you give me some advice, how should I configure vscode setting.json and launch.json .

Is it because of relying on third-party libraries, such as: openzeppelin?
==============================================================

Thanks to @acuarica for leading me here, looks like this is a tough question, above is my original question, hopefully we can find the answer here!

**Comment by pupudaye** [comment]
This error occurs more often, of course, it is not really impossible to connect, the network is always available.

“Could not connect to your Ethereum client.
Please check that your Ethereum client:
- is running
- is accepting RPC connections (i.e., "--rpc" or "--http" option is used in geth)
- is accessible over the network
- is properly configured in your Truffle configuration file (truffle-config.js)
”

**Comment by xhulz** [comment]
PR ref: #229
